### PR TITLE
Add block compaction related stats

### PR DESCRIPTION
**Pull request by wenleix**

### CODE REVIEW

**Comment by wenleix** [comment]
Deployed in test cluster. It shows all the compactions are skipped with @highker's test query (concurrency = 5)

screen shot 2017-11-30 at 22 25 31

**Comment by wenleix** [comment]
I have tested in a test cluster with a few randomly selected chosen high memory queries. But it shows all of the compactions are skipped. It might be workload specific.

I would like to get this merged so we can collect stats from production.

One issue is this adds 26 CounterStats. We can reduce it by half by only reporting the bytes (but not number of blocks). We can also make less saving buckets. 5 buckets seems to be enough.

Would like to hear comments. Especially suggestions on stats name are welcomed.  @haozhun , @highker

File: presto-main/src/main/java/com/facebook/presto/operator/PagesIndexStats.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+import io.airlift.stats.CounterStat;
+import org.weakref.jmx.Managed;
+import org.weakref.jmx.Nested;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public class PagesIndexStats
+{
+    private final CounterStat totalBlocks = new CounterStat();
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/operator/PagesIndexStats.java]: `CounterStat` is quite expensive. I was working on an improvement (https://github.com/highker/airlift/tree/decay) but can't convince myself the improvement is a win. How about just use atomic data types? @martint

**Comment by wenleix** [comment]
Alternatively we can just store 3 types of savings:
- Small: < 10% saving
- Medium: 10% - 30% saving
- Large: > 30% saving

File: presto-main/src/main/java/com/facebook/presto/operator/PagesIndexStats.java (hunk 1 above)
Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/operator/PagesIndexStats.java]: @highker : Yep. A atomic long is (mostly) enough in this case.

**Comment by haozhun** [comment]
I'm un-assigning myself given the age of this PR, and that comments from fellow reviewers were not addressed.

**Comment by wenleix** [comment]
@haozhun : The comment from fellow reviewer is a minor issue. The real question is should we export all 10% - 90% stats (since that makes the code quite long and not easy to follow), or just export 3 of them.

**Pull request closed without merging**